abd-clam/src/core/cluster/tree.rs

```diff
@@ -72,4 +78,372 @@ impl<I: Instance, U: Number, D: Dataset<I, U>> Tree<I, U, D> {
+            let leaf_history = Cluster::<U>::name_to_history(&leaf);
+
+            // We start from index 1 to skip the identically 1 prefix at index 0
+            for step in 1..leaf_history.len() {
+                let branch = leaf_history[step];
+
+                // If we don't have any children here, we need to build them out
+                if cur.children.is_none() {
+                    // Construct the names for the left and right children
+                    let mut left_history = leaf_history[0..step].to_vec();
+                    left_history.push(false);
+
+                    let mut right_history = leaf_history[0..step].to_vec();
+                    right_history.push(true);
+
+                    let left_name = Cluster::<U>::history_to_name(&left_history);
+                    let right_name = Cluster::<U>::history_to_name(&right_history);
+
+                    // Deserialize the left and right child clusters
+                    let left: Cluster<U> = recover_serialized_cluster(cluster_path.join(left_name))?;
+                    let right: Cluster<U> = recover_serialized_cluster(cluster_path.join(right_name))?;
+
+                    // Get the childinfo (arg_l, arg_r, etc.)
+                    let parent_name = Cluster::<U>::history_to_name(&leaf_history[0..step]);
+                    let childinfo = recover_serialized_childinfo(childinfo_path.join(&parent_name))?;
+
+                    // Reconstruct the children
+                    cur.children = Some(Children {
+                        left: Box::new(left),
+                        right: Box::new(right),
+                        arg_l: childinfo.arg_l,
+                        arg_r: childinfo.arg_r,
+                        polar_distance: <U as Number>::from_le_bytes(&childinfo.polar_distance_bytes),
+                    });
+                }
+
+                // Unwrap is justified here because we always check if cur.children is None and
+                // load in the children if it is.
+                #[allow(clippy::unwrap_used)]
+                let children = cur.children.as_mut().unwrap();
+
+                // Choose which branch to take
+                if branch {
+                    cur = &mut children.right;
+                } else {
+                    cur = &mut children.left;
+                }
+            }
+        }
+
+        let root = *boxed_root;
+
+        Ok(Self {
+            data: dataset,
+            depth: root.max_leaf_depth(),
+            root,
+            _i: PhantomData,
+        })
+    }
+}
+
+/// Serializes a serializeable (`impl Serialize`) object to a given path
+///
+/// # Errors
+/// This function will error out on serialization or file i/o errors.
+fn serialize_to_file<S: Serialize>(path: PathBuf, object: &S) -> Result<(), String> {
+    let mut file = File::create(path).map_err(|e| e.to_string())?;
+    let info_string = serde_json::to_string(&object).map_err(|e| e.to_string())?;
```

Comment:
Add a TODO here to encode the `object` as bytes instead of `json`. We'll add the byte-encodings later.